src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -156,4 +161,26 @@ public static ValidationException validateKnnField(IndexMetadata indexMetadata,
 
         return null;
     }
+
+    /**
+     * Gets the load time parameters for a given engine.
+     *
+     * @param spaceType Space for this particular segment
+     * @param knnEngine Engine used for the native library indices being loaded in
+     * @param indexName Name of OpenSearch index that the segment files belong to
+     * @return load parameters that will be passed to the JNI.
+     */
+    public static Map<String, Object> getLoadParameters(SpaceType spaceType, KNNEngine knnEngine, String indexName) {
+        Map<String, Object> loadParameters = Maps.newHashMap(ImmutableMap.of(
```

Comment:
Are we going to mutate this map later? Seems that at the end we return mutable object from this method. 
```suggestion
        final Map<String, Object> loadParameters;
        if (KNNEngine.NMSLIB.equals(knnEngine)) {
              loadParameters = ImmutableMap.of(
                            SPACE_TYPE, spaceType.getValue(),
                            HNSW_ALGO_EF_SEARCH, KNNSettings.getEfSearchParam(indexName)
              );
        } else {
              loadParameters = ImmutableMap.of(SPACE_TYPE, spaceType.getValue());
        }
        return loadParameters;
```

Reply:
or we can just return UnModifiableMap.

Reply:
Prefer the unmodifiable map approach. Will go with that.